eigrpd/eigrp_packet.c

```diff
@@ -1154,25 +1154,56 @@ u_int16_t eigrp_add_internalTLV_to_stream(struct stream *s,
 	u_int16_t length;
 
 	stream_putw(s, EIGRP_TLV_IPv4_INT);
-	if (pe->destination->prefixlen <= 8) {
-		stream_putw(s, 0x001A);
-		length = 0x001A;
-	}
-	if ((pe->destination->prefixlen > 8)
-	    && (pe->destination->prefixlen <= 16)) {
-		stream_putw(s, 0x001B);
-		length = 0x001B;
-	}
-	if ((pe->destination->prefixlen > 16)
-	    && (pe->destination->prefixlen <= 24)) {
-		stream_putw(s, 0x001C);
-		length = 0x001C;
-	}
-	if (pe->destination->prefixlen > 24) {
-		stream_putw(s, 0x001D);
-		length = 0x001D;
+	switch (pe->destination->prefixlen) {
```

Comment:
we can calculate the number of bytes we need from the prefixlen with the following formula: `((pe->destination->prefixlen - 1) / 8) + 1`

the formula for ipv6 prefixes is a bit different: https://github.com/openbsd/src/blob/master/usr.sbin/eigrpd/eigrpe.h#L72

Reply:
This code is v4 specific at this point in time.  Additionally the code makes some assumptions about how it calculates the packet length that I would prefer to not have to think through at this point in time.  Especially since I want to be able to get in split-horizon code here next.  I promise it is on my radar though :).  I've also squashed the 3rd commit into the 2nd since it just fixes an issue there.